src/swarm/neo/connection/ConnectionBase.d

```diff
@@ -156,6 +189,11 @@ abstract class ConnectionBase: ISelectClient
                 if ( this.idle )
                 {
                     this.resume(this.fiber_token.get(), this.outer);
+
+                    // If the send loop exited, notify the calling fiber. (See
+                    // note on error handling, above.)
+                    if ( !this.loop_started )
+                        throw send_loop_exited_ex;
```

Comment:
Hm, wouldn't it make more sense to throw the exception which cause the send loop to exit?

Reply:
Yeah... We'd have to dup it I guess...? By the time we get to this point, the exception no longer exists.

Reply:
Updated. (Retested with https://github.com/sociomantic-tsunami/dhtnode/issues/13.)

Reply:
For the record, the code has already been updated: Making `send_loop_exited_ex` non-static, assigning the exception caught in `fiberMethod` and rethrowing it here is sufficient. The exception doesn't need to be dupped.

Reply:
Two more things:
- `throw this.send_loop_exited_ex;`
- Reset `this.send_loop_exited_ex = null` after throwing (e.g. in a `scope (exit)` clause).

Reply:
Updated. (Also added an assert that the exception is non-null before throwing it.)